### ISSUE TITLE
AWS Parameter Store provider docs mention IAM policy actions that do not exist

### ISSUE THREAD

**Issue by isker** [external-secrets/external-secrets#2043]
https://external-secrets.io/v0.7.2/provider/aws-parameter-store/ gives this example policy document:
```
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Effect": "Allow",
      "Action": [
        "ssm:GetParameterWithContext",
        "ssm:ListTagsForResourceWithContext",
        "ssm:DescribeParametersWithContext",
      ],
      "Resource": "arn:aws:ssm:us-east-2:1234567889911:parameter/dev-*"
    }
  ]
}
```

`ssm:GetParameterWithContext` [does not exist](https://docs.aws.amazon.com/search/doc-search.html?searchPath=documentation-guide&searchQuery=GetParameterWithContext&this_doc_product=AWS%20Systems%20Manager&this_doc_guide=API%20Reference&facet_doc_product=AWS%20Systems%20Manager&facet_doc_guide=API%20Reference); likewise for the rest.  I think these may have been copy/pasted from function names in the [go SDK](https://pkg.go.dev/github.com/aws/aws-sdk-go/service/ssm#SSM.GetParameterWithContext).  Such functions are specific to go and aren't a part of the ssm API.  The docs should mention actions that exist.

**Comment by moolen** [comment]
Hey, thanks for reporting! This has been fixed in #1949 and is already on `main` but hasn't been released yet.

**Comment by isker** [comment]
Whoops.  I was going to verify a fix myself locally and then open a PR.  Thanks!
